lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/python
+# (c) 2017, Nathan Davison <ndavison85@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: listen_ports_facts
+
+author:
+    - Nathan Davison (@ndavison)
+
+version_added: "2.6"
+
+description:
+    - Gather facts on processes listening on TCP and UDP ports. Optionally provide whitelists to gather facts on violations of the whitelist.
+
+short_description: Gather facts on processes listening on TCP and UDP ports.
+
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP port numbers that are expected to have processes listening.
+    default: []
+  whitelist_udp:
+    description:
+      - A list of UDP port numbers that are expected to have processes listening.
+    default: []
+'''
+
+EXAMPLES = '''
+- name: gather facts on listening ports, and populate the tcp_listen_violations fact with processes listening on TCP ports that are not 22, 80, or 443.
```

Comment:
Description should be capitalized.
```suggestion
- name: Gather facts on listening ports, and populate the tcp_listen_violations fact with processes listening on TCP ports that are not 22, 80, or 443.
```